### PR TITLE
allow topic searches to continue past invalid multi-type topics.

### PR DESCRIPTION
**Pull request by rjb0026**
**Public API Changes**
None

**Description**
Fix bug that causes desired image topics to sometimes not stream correctly

https://github.com/RobotWebTools/web_video_server/issues/145

### CODE REVIEW

**Comment by rjb0026** [comment]
@EzraBrooks I see that you reviewed another recently closed PR on this project. Are you a maintainer? Or do you know who is? I'm looking for someone to review and test this small PR, but I have no idea who maintains this project.

**Comment by EzraBrooks** [comment]
Yes, I am a maintainer, though I have less experience with this particular repo than with some others like roslibjs.

**Review by EzraBrooks — APPROVED** [review]
This change seems sensical to me. pinging @sea-bass in case he has thoughts

File: src/image_streamer.cpp [hunk 1]
```diff
@@ -39,7 +39,7 @@ void ImageTransportImageStreamer::start()
   for (auto topic_and_types : tnat) {
     if (topic_and_types.second.size() > 1) {
       // explicitly avoid topics with more than one type
-      break;
+      continue;
```
Comment (sea-bass) [inline src/image_streamer.cpp]: the comment should be updated, then

File: src/web_video_server.cpp [hunk 2]
```diff
@@ -193,7 +193,7 @@ bool WebVideoServer::handle_stream(const async_web_server_cpp::HttpRequest &requ
       for (auto topic_and_types : tnat) {
         if (topic_and_types.second.size() > 1) {
           // explicitly avoid topics with more than one type
-          break;
+          continue;
```
Comment (sea-bass) [inline src/web_video_server.cpp]: same for the rest

**Review by sea-bass — CHANGES_REQUESTED** [review]
seems fine, but please update the comments which still say "explicitly avoid", since we are not doing that anymore

File: src/image_streamer.cpp (hunk 1 above)
Comment (rjb0026) [reply src/image_streamer.cpp]: So this does still skip past and avoid multi-type topics, it just doesn't completely stop the topic search for other single-type topics if you happen to encounter a multi-type topic. Would it be better to phrase it as:
"avoid topics with more than one type" or maybe
"skip over topics with more than one type" ?

Comment (rjb0026) [reply src/image_streamer.cpp]: Changed it slightly to "skip over topics with more than one type", which I felt was a little more descriptive of what's actually happening.

**Review by sea-bass — APPROVED** [review]
Thank you!

**Pull request merged**